server.c

```diff
@@ -2319,6 +2378,10 @@ server_reload(struct nsd *nsd, region_type* server_region, netio_type* netio,
 	/* Restart dumping stats if required.  */
 	time(&nsd->st->boot);
 	set_bind8_alarm(nsd);
+	/* Switch to a different set of stat array for new server processes,
+	 * because they can briefly coexist with the old processes. They
+	 * have their own stat structure. */
+	nsd->stat_current = (nsd->stat_current==0?1:0);
```

Comment:
You could write this as: `!nsd->stat_current`(?)

Reply:
I think that is true, but may depend on compiler version and language specification. So I prefer to be explicit about it here.